File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

pair<string, string> convBin(int num1, int num2, int n) {
    string s1 = "";
    string s2 = "";

    while (num1 != 0) {
        s1 = to_string(num1 % 2) + s1;
        num1 /= 2;
    }

    while (num2 != 0) {
        s2 = to_string(num2 % 2) + s2;
        num2 /= 2;
    }

    while (s1.length() < n) {
        s1 = "0" + s1;
    }
    
    while (s2.length() < n) {
        s2 = "0" + s2;
    }
    return {s1, s2};
}

vector<string> solution(int n, vector<int> arr1, vector<int> arr2) {
    vector<string> grid(n); // 지도를 만들 공간 생성
    
    for(int i = 0; i < n; i++){
        int a = arr1[i];
        int b = arr2[i];
        pair<string, string>res = convBin(a, b, n);
        
        string row = "";
        for(int j = 0; j < n; j++){
            if (res.first[j] == '1' || res.second[j] == '1'){
                row += '#';
            }
            else row += ' ';
        }
        grid[i] = row;
    }
    return grid;
}

File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.py
def convBin(num1, num2, n):
    s1 =""
    s2 = ""
    
    while(num1 != 0):
        s1 = str(num1 % 2) + s1
        num1 //= 2
    
    while(num2 != 0):
        s2 = str(num2 % 2) + s2
        num2 //= 2
    
    # zero padding
    s1 = "0" * (n - len(s1)) + s1
    s2 = "0" * (n - len(s2)) + s2
    
    return (s1, s2)

def solution(n, arr1, arr2):
    answer = []
    for i in range(n):
        a, b = arr1[i], arr2[i]
        sa, sb = convBin(a, b, n)
        tmp = ""
        for j in range(n):
            if sa[j] == '1' or sb[j] == '1':
                tmp += "#"
            else:
                tmp += " "
        answer.append(tmp)
        
    return answer